### PR TITLE
First draft of the data-storage-scheme

### PR DESCRIPTION
**Pull request by oed**
This is a first draft of the data structure of the data store.

### CODE REVIEW

File: 3BOX-JS-DATA-STRUCTURE.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+TODO - Christian: add the 3box profile schema here.
+
+### Private data store
+The private data store is an orbit-db KV-store with additional encryption. The encryption scheme for adding a key-value entry would work as follows:
+
+* Generate a random salt and store it under the `3BOX_SALT` key.
+* Compute the `key` by taking `h(PLAIN_KEY | salt)`
+* Generate a random nonce `n`
+* Compute the ciphertext: `nacl.secretbox(PLAIN_VALUE, n, secretKey)`
+* Encode `value` as `Base64(n).Base64(ciphertext)`
```
[the remark this answers is not in the document]
Comment (oed) [reply 3BOX-JS-DATA-STRUCTURE.md]: Base64 doesn't include the `.` character right? So I think it should be no problem. Do you have any other suggestions? I don't think orbitdb has any preferences, just any string.

File: 3BOX-JS-DATA-STRUCTURE.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Public profile
+The public profile is an IPLD formated ipfs object that contains the public information about the user such as name and picture.
+
+TODO - Christian: add the 3box profile schema here.
+
+### Private data store
+The private data store is an orbit-db KV-store with additional encryption. The encryption scheme for adding a key-value entry would work as follows:
+
+* Generate a random salt and store it under the `3BOX_SALT` key.
```
[the remark this answers is not in the document]
Comment (oed) [reply 3BOX-JS-DATA-STRUCTURE.md]: yes, thanks!

File: 3BOX-JS-DATA-STRUCTURE.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+* Compute the `key` by taking `h(PLAIN_KEY | salt)`
+* Generate a random nonce `n`
+* Compute the ciphertext: `nacl.secretbox(PLAIN_VALUE, n, secretKey)`
```
[the remark this answers is not in the document]
Comment (oed) [reply 3BOX-JS-DATA-STRUCTURE.md]: Yes we need to! Totally forgot about that.

File: 3BOX-JS-DATA-STRUCTURE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (oed) [reply 3BOX-JS-DATA-STRUCTURE.md]: Seems like orbitdb accepts js objects as value. Changing to reflect that.

**Comment by oed** [comment]
@christianlundkvist One very good reason for using orbitdb for the profile as well is that we can handle it being changed in multiple locations since with the CRDT structure we can always sync the changes. But I guess what we have now is fine for the MVP.

**Comment by christianlundkvist** [comment]
Merging this now, since it’s complete with Profile schema.

**Pull request merged**